graph/src/data/query/error.rs

```diff
@@ -263,19 +263,23 @@ impl From<StoreError> for QueryExecutionError {
 #[derive(Debug)]
 pub enum QueryError {
     EncodingError(FromUtf8Error),
-    ParseError(q::ParseError),
+    ParseError(failure::Error),
```

Comment:
Switched this to `anyhow`, and wrapped it in an `Arc` which I've learned is the correct way of making an error clone.

Reply:
I see that nice. Where did you learn this? I've been wondering about the right way to deal with error clone. Does `Arc<failure::Error>` impl `std::Error::Error` or what's the deal there?

Reply:
I learned that here https://github.com/dtolnay/anyhow/issues/7. Actually `anyhow::Error` itself doesn't impl `Error`, but it can be converted into a `Box<dyn Error>` if necessary.